src/api_pull/api_puller.R

```diff
@@ -0,0 +1,74 @@
+# to do (j): document this function
+api_puller <- function(site, start_dt, end_dt, api_token, dump_dir) {
+
+  locs <- hv_locations_all(api_token)
+
+  # make a list of site names
+
+  options(scipen = 999)
+
+  site_loc <- locs %>%
+    mutate(name = tolower(name)) %>%
+    filter(grepl(site, name))
+
+  site_loc_list <- site_loc$id
+
+  # Get data for each site location. Note this maps over the entire list of locations,
+  # many of which are unlikely to be active during the time you specify. Don't freak out if
+  # you see a bunch of '404 Not Found' errors, you're just seeing the list of locations
+  # that are not active. The data frame 'alldata' should contain your data from all applicable
+  # sites during the time frame indicated. Note that this will take some time (one month of
+  # data for 5 sites takes ~10 mins. Be patient!
+
+  # Add date range you are interested in; data are stored in HydroVu in UTC
+  # Here, a way to find the most recent download of the data. Use this as the start date to
+  # reduce overlapping data
+
+  # tz weirdness
+  utc_start_date <- format(as.POSIXct(format(start_dt, tz = "UTC"), tz = "UTC"), format = "%Y-%m-%d %H:%M:%S")
+
+  utc_end_date <- format(as.POSIXct(format(end_dt, tz = "UTC"), tz = "UTC"), format = "%Y-%m-%d %H:%M:%S")
+
+  timezone = "UTC"
+
```

Comment:
might be nice to add something like `print(paste0("No data at ", site, " during this time frame"))` for testing/checking purposes.